app.go

```diff
@@ -89,15 +93,23 @@ func (po provideOption) String() string {
 }
 
 // Invoke registers functions that are executed eagerly on application start.
-// Arguments for these functions are provided from the application's
-// dependency injection container.
+// Arguments for these invocations are built using the constructors registered
+// by Provide. Passing multiple Invoke options appends the new invocations to
+// the application's existing list.
+//
+// Unlike constructors, invocations are always executed, and they're always
+// run in order. Invocations may have any number of returned values. If the
+// final returned object is an error, it's assumed to be a success indicator.
+// All other returned values are discarded.
 //
-// Unlike constructors, invoked functions are always executed, and they're
-// always run in order. Invoked functions may have any number of returned
-// values. If the final returned object is an error, it's assumed to be a
-// success indicator. All other returned values are discarded.
+// Typically, invoked functions take a handful of high-level objects (whose
+// constructors depend on lower-level objects) and introduce them to each
```

Comment:
`and introduce them to each other` not sure what is meant by this

Reply:
Happy to come back to the wording here as necessary; this proved much clearer in our internal "intro to Fx" documentation. This means, very broadly, that `Invoke` is typically used to take two types and make them interact with each other somehow. Often, they're taking the top-level user-defined objects (like HTTP handlers) and making them interact with the top-level type provided by some shared library (like a mux that's already bound to a port, with some common middleware applied).